checker/jdk/nullness/src/javax/script/CompiledScript.java

```diff
@@ -0,0 +1,121 @@
+     *
+     * @param context A <code>ScriptContext</code> that is used in the same way as
+     * the <code>ScriptContext</code> passed to the <code>eval</code> methods of
+     * <code>ScriptEngine</code>.
+     *
+     * @return The value returned by the script execution, if any.  Should return <code>null</code>
+     * if no value is returned by the script execution.
+     *
+     * @throws ScriptException if an error occurs.
+     * @throws NullPointerException if context is null.
+     */
+
+    public abstract @Nullable Object eval(ScriptContext context) throws ScriptException;
+
+    /**
+     * Executes the program stored in the <code>CompiledScript</code> object using
+     * the supplied <code>Bindings</code> of attributes as the <code>ENGINE_SCOPE</code> of the
+     * associated <code>ScriptEngine</code> during script execution.  If bindings is null,
+     * then the effect of calling this method is same as that of eval(getEngine().getContext()).
+     * <p>.
+     * The <code>GLOBAL_SCOPE</code> <code>Bindings</code>, <code>Reader</code> and <code>Writer</code>
+     * associated with the default <code>ScriptContext</code> of the associated <code>ScriptEngine</code> are used.
+     *
+     * @param bindings The bindings of attributes used for the <code>ENGINE_SCOPE</code>.
+     *
+     * @return The return value from the script execution
+     *
+     * @throws ScriptException if an error occurs.
+     */
+    @SuppressWarnings("nullness") // as getEngine()'s implementation will never be returning null and same applies for getContext() method in ScriptEngine.java as specified in documentation. 
+    public Object eval(@Nullable Bindings bindings) throws ScriptException {
+
+        ScriptContext ctxt = getEngine().getContext();
+
+        if (bindings != null) {
+            SimpleScriptContext tempctxt = new SimpleScriptContext();
+            tempctxt.setBindings(bindings, ScriptContext.ENGINE_SCOPE);
+            tempctxt.setBindings(ctxt.getBindings(ScriptContext.GLOBAL_SCOPE),
+                    ScriptContext.GLOBAL_SCOPE);
+            tempctxt.setWriter(ctxt.getWriter());
+            tempctxt.setReader(ctxt.getReader());
+            tempctxt.setErrorWriter(ctxt.getErrorWriter());
+            ctxt = tempctxt;
+        }
+
+        return eval(ctxt);
+    }
+
+
+    /**
+     * Executes the program stored in the <code>CompiledScript</code> object.  The
+     * default <code>ScriptContext</code> of the associated <code>ScriptEngine</code> is used.
+     * The effect of calling this method is same as that of eval(getEngine().getContext()).
+     *
+     * @return The return value from the script execution
+     *
+     * @throws ScriptException if an error occurs.
+     */
+    @SuppressWarnings("nullness")    // as getEngine()'s implementation will never be returning null and same applies for getContext() method in ScriptEngine.java as specified in documentation.  
```

Comment:
If the two methods are already correctly annotated, then why is any warning suppression needed?

Reply:
getEngine().getContext() value return is @nullable type according to checker. But specifications disagrees with it.